CHANGELOG.md

```diff
@@ -1,3 +1,56 @@
+  time during failed retries of network requests in `olp::network::client::OlpClient` class (both sync and async versions) in 
+  a way that the accumulated wait time during retries shouldn't be longer 
+  than `RetrySettings.timeout` property.
+* Added `curl_global_init()` and `curl_global_cleanup()` to `olp::http::NetworkCurl` class (Linux default Network implementation). 
+  Please refer to `olp::client::OlpClientSettingsFactory::CreateDefaultNetworkRequestHandler()` function 
+  for more infromation.
+* Fixed various warnings (MSVC, Clang, GCC).
+* Fixed of thread concurrency problem in `olp::client::OlpClient` class which in some situations resulted in crash.
+* Reduced data copy in SerializerWrapper.h which will increase performance.
+
+**olp-cpp-sdk-authentication**
+
+* Added a new API to authenticate OLP with a custom request body.
+* Added a new API `olp::authentication::AuthenticationClient::IntrospectApp()` method which can be used to retrieve information
+  about the application associated with access token.
+* Added new `olp::authentication::AuthenticationClient` class constructor.
+  This is a part of authentication refactoring. 
+  Added new `olp::authentication::AuthenticationClient` constructor using `olp::authentication::AuthenticationSewttings` to setup network and 
+  task scheduler. Our goal is to remove setters, users should switch to this constructor, 
+  we aim for decoupling this from the `olp::authentication::TokenEndpoint` in the near future.
+
+**olp-cpp-sdk-dataservice-read**
+
+* Added methods to `olp::dataservice::read::VersionedLayerClient` class that supports getting multiple partitions at once.
+* Added Seek methods for `olp::dataservice::read::StreamLayerClient`.
+* Added Poll method to `olp::dataservice::read::StreamLayerClient`.
+* Added TileRequest class and corresponding methods to `olp::dataservice::read::VersionedLayerClient` interface. 
+  This will make possible to get data using tile key.
+* Added additional fields to `olp::dataservice::read::PartitionsRequest` class. This gives user access to additional metadata
+  fields: 
+  - data size
+  - compressed data size
+  - checksum
+  - crc
+* Deprecated WithVersion methods in request classes.
+  `olp::dataservice::read::VersionedLayerClient` needs to serve consistent data for all requests which are triggered on a 
+  particular instance. If user do not know the catalog version up front and always use boost::none 
+  for version, the risk of returning different catalog versions data is high. 
+  Therefore we lock the catalog version in the constructor of the `olp::dataservice::read::VersionLayerClient` 
+  instance and all requests are therefore served with that particular catalog version. 
+  Hence we do not need the WithVersion() methods in the requests structures anymore and they 
+  will be ignore during requests and the constructor passed catalog version will be taken instead.
```

Comment:
```suggestion
 `olp::dataservice::read::VersionedLayerClient` needs to provide consistent data for all requests that are triggered in 
 every particular instance. If you do not know the catalog version and use `boost::none` instead of the version number, you can get data from 
 a different catalog version. 
 Therefore, we locked the catalog version in the constructor of the `olp::dataservice::read:: VersionLayerClient` instance. 
 Now, all requests get that particular catalog version. 
 Hence, we do not need the `WithVersion(`) methods in the request structures anymore. The versions are now ignored 
 in requests, and the version passed in the constructor is used instead.
```